File: RiotShopBackEnd/RiotShopBackEnd/Models/ProductTemplate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RiotShopBackEnd.Models;

[Table("Product_Templates")]
public class ProductTemplate
{
    [Key]
    [Column("template_id")]
    public int TemplateId { get; set; }

    [Required]
    [Column("game_id")]
    public int GameId { get; set; }

    [Required]
    [MaxLength(255)]
    [Column("title")]
    public string Title { get; set; } = string.Empty;

    [Column("description")]
    public string? Description { get; set; }

    [Required]
    [Column("base_price", TypeName = "decimal(10,2)")]
    public decimal BasePrice { get; set; }

    [Column("is_featured")]
    public bool IsFeatured { get; set; } = false;

    [MaxLength(100)]
    [Column("tag_rank")]
    public string? TagRank { get; set; }

    [MaxLength(100)]
    [Column("tag_skins")]
    public string? TagSkins { get; set; }

    [MaxLength(100)]
    [Column("tag_collection")]
    public string? TagCollection { get; set; }

    [MaxLength(500)]
    [Column("image_url")]
    public string? ImageUrl { get; set; }

    // Navigation property
    [ForeignKey("GameId")]
    public GameType? GameType { get; set; }
}

File: riotshop/riotshop/app/src/main/java/com/example/riotshop/models/ProductTemplate.java
package com.example.riotshop.models;

import com.google.gson.annotations.SerializedName;

public class ProductTemplate {
    @SerializedName("templateId")
    private int templateId;
    
    @SerializedName("gameId")
    private int gameId;
    
    @SerializedName("title")
    private String title;
    
    @SerializedName("description")
    private String description;
    
    @SerializedName("basePrice")
    private double basePrice;
    
    @SerializedName("isFeatured")
    private boolean isFeatured;
    
    @SerializedName("tagRank")
    private String tagRank;
    
    @SerializedName("tagSkins")
    private String tagSkins;
    
    @SerializedName("tagCollection")
    private String tagCollection;
    
    @SerializedName("imageUrl")
    private String imageUrl;
    
    @SerializedName("gameName")
    private String gameName; // Backend có thể trả về gameName trực tiếp
    
    @SerializedName("gameType")
    private GameType gameType;
    
    @SerializedName("inventory")
    private Inventory inventory;

    // Getters
    public int getTemplateId() {
        return templateId;
    }

    public int getGameId() {
        return gameId;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public double getBasePrice() {
        return basePrice;
    }

    public boolean isFeatured() {
        return isFeatured;
    }

    public String getTagRank() {
        return tagRank;
    }

    public String getTagSkins() {
        return tagSkins;
    }

    public String getTagCollection() {
        return tagCollection;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getGameName() {
        if (gameName != null && !gameName.isEmpty()) {
            return gameName;
        }
        if (gameType != null) {
            return gameType.getName();
        }
        return "";
    }

    public GameType getGameType() {
        return gameType;
    }
    
    public Inventory getInventory() {
        return inventory;
    }
    
    // Inner class for Inventory
    public static class Inventory {
        @SerializedName("packageId")
        private int packageId;
        
        @SerializedName("quantityAvailable")
        private int quantityAvailable;
        
        @SerializedName("price")
        private double price;
        
        @SerializedName("lastUpdated")
        private String lastUpdated;
        
        public int getPackageId() {
            return packageId;
        }
        
        public int getQuantityAvailable() {
            return quantityAvailable;
        }
        
        public double getPrice() {
            return price;
        }
        
        public String getLastUpdated() {
            return lastUpdated;
        }
    }
}

